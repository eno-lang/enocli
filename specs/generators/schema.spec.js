const path = require('path');

const analyze = require('../../lib/analyze.js');
const generateSchema = require('../../lib/generators/schema.js');

const DOCUMENT_PATH = path.join(__dirname, '../fixtures/document.eno');
const SECTIONS_DOCUMENT_PATH = path.join(__dirname, '../fixtures/sections.eno');

describe('generateSchema', () => {
  it('generates a schema from an abstract syntax tree', () => {
    const astDocument = analyze(DOCUMENT_PATH);
    const schema = generateSchema(astDocument);

    expect(schema).toMatchSnapshot();
  });

  it('correctly exports section depths', () => {
    const astDocument = analyze(SECTIONS_DOCUMENT_PATH);
    const schema = generateSchema(astDocument);

    expect(schema).toMatchSnapshot();
  });
});
