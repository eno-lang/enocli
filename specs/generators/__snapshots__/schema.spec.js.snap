// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSchema correctly exports section depths 1`] = `
"@element required
@key string section_depth_1
# section (depth: 1)
@element required
@key string section_depth_2
## section (depth: 2)
@element required
@key string section_depth_3
### section (depth: 3)

@element required
@key string section_depth_1_dup
# section (depth: 1)
@element required
@key string section_depth_2
## section (depth: 2)
@element required
@key string section_depth_3
### section (depth: 3)

@element required
@key string section_depth_2_dup
## section (depth: 2)
@element required
@key string section_depth_3
### section (depth: 3)"
`;

exports[`generateSchema generates a schema from an abstract syntax tree 1`] = `
"@element required
@key string empty
empty

@element required
@key string field
@value required string
field: value

@element required
@key string fieldset
fieldset:

@element required
@key string entry
@value required string
entry: value

@element required
@key string list
@values required string
list:

@element required
@key string section
# section
@element required
@key string field
@value required string
field: value"
`;
